src/hera/workflows/user_container.py

```diff
@@ -13,16 +13,82 @@
 from hera.workflows.resources import Resources
 from hera.workflows.volume import _BaseVolume
 
+# TODO: Mixins imports user container, so this avoids a circular import. Fix in a future release
```

Comment:
mixins import the user container unfortunately. Duplicating here means we don't introduce breaking changes

Reply:
Looking at [the template mixin](https://github.com/argoproj-labs/hera/blob/961963ba6c2866006ebd850e3c128d81a596ad17/src/hera/workflows/_mixins.py#L413-L448), I'm not sure initially how to even solve this as it's used for attribute types 🤔

Reply:
Naively, we could declare UserContainer as a private attr of mixins then export it in user_container as a public class